src/Files.App/Utils/Storage/Helpers/SyncRootHelpers.cs

```diff
@@ -1,144 +1,61 @@
+			using ComPtr<IStorageProviderStatusUISource> pStorageProviderStatusUISource = default;
+			using ComPtr<IStorageProviderStatusUI> pStorageProviderStatusUI = default;
+			using ComPtr<IStorageProviderQuotaUI> pStorageProviderQuotaUI = default;
 
-		private static unsafe (bool Success, ulong Capacity, ulong Used) GetSyncRootQuotaFromSyncRootId(string syncRootId)
-		{
-			RegistryKey? key;
-			if ((key = Registry.LocalMachine.OpenSubKey($"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\SyncRootManager\\{syncRootId}")) is null)
-			{
+			var hr = PInvoke.CoCreateInstance(
+				&factoryClsid,
+				null,
+				CLSCTX.CLSCTX_LOCAL_SERVER,
+				(Guid*)Unsafe.AsPointer(ref Unsafe.AsRef(in IStorageProviderStatusUISourceFactory.Guid)),
+				(void**)pStorageProviderStatusUISourceFactory.GetAddressOf());
+			if (hr.Failed)
 				return (false, 0, 0);
-			}
 
-			using (key)
+			var syncRootIdHString = new MarshalString.Pinnable(syncRootId);
+			fixed (char* pSyncRootIdHString = syncRootIdHString)
 			{
-				if (key.GetValue("StorageProviderStatusUISourceFactory") is string statusUIclass)
-				{
-					StorageProviderStatusUI statusUI;
-					using (ComPtr<IStorageProviderStatusUISourceFactory> sourceFactoryNative = default)
-					{
-						Guid statusUIclassGuid = Guid.Parse(statusUIclass);
-						if (PInvoke.CoCreateInstance(&statusUIclassGuid, null, Windows.Win32.System.Com.CLSCTX.CLSCTX_LOCAL_SERVER, (Guid*)Unsafe.AsPointer(ref Unsafe.AsRef(in IStorageProviderStatusUISourceFactory.Guid)), (void**)sourceFactoryNative.GetAddressOf()) != 0)
-						{
-							return (false, 0, 0);
-						}
-
-						MarshalString.Pinnable syncRootIdHstring = new(syncRootId);
-						fixed (char* ptr = syncRootIdHstring)
-							using (ComPtr<IStorageProviderStatusUISource> sourceNative = default)
-							{
-								ExceptionHelpers.ThrowExceptionForHR(sourceFactoryNative.Get()->GetStatusUISource(syncRootIdHstring.GetAbi(), sourceNative.GetAddressOf()));
+				hr = pStorageProviderStatusUISourceFactory.Get()->GetStatusUISource(syncRootIdHString.GetAbi(), pStorageProviderStatusUISource.GetAddressOf()).ThrowOnFailure();
+				hr = pStorageProviderStatusUISource.Get()->GetStatusUI(pStorageProviderStatusUI.GetAddressOf()).ThrowOnFailure();
+				hr = pStorageProviderStatusUI.Get()->GetQuotaUI(pStorageProviderQuotaUI.GetAddressOf()).ThrowOnFailure();
 
-								using (ComPtr<IStorageProviderStatusUI> statusNative = default)
-								{
-									ExceptionHelpers.ThrowExceptionForHR(sourceNative.Get()->GetStatusUI(statusNative.GetAddressOf()));
-									statusUI = StorageProviderStatusUI.FromAbi((nint)statusNative.Get());
-								}
-							}
-					}
-					return (true, statusUI.QuotaUI.QuotaTotalInBytes, statusUI.QuotaUI.QuotaUsedInBytes);
-				}
-				else
-				{
-					return (false, 0, 0);
-				}
+				hr = pStorageProviderQuotaUI.Get()->GetQuotaTotalInBytes(&ulTotalSize);
+				hr = pStorageProviderQuotaUI.Get()->GetQuotaUsedInBytes(&ulUsedSize);
```

Comment:
I think we should return false on these instead of throwing or ignoring the error.

Reply:
Done.